File: cpp/31e_80A_PanoramixsPrediction.cpp
#include<bits/stdc++.h>

bool isprime(int a){
    for(int i = 2; i < ((a / 2) + 1); i++){
        if(a % i == 0){
            return false;
        }
    }
    return true;
}

int main(){
    int n, m;
    std::cin>>n>>m;
    if(!isprime(m)){
        std::cout<<"NO";
    }
    else{
        int flag = 0;
        for(int i = n + 1; i < m; i++){
            if(isprime(i)){
                flag = 1;
                break;
            }
        }
        if(flag == 0){
            std::cout<<"YES";
        }
        else{
            std::cout<<"NO";
        }
    }
    return 0;
}

File: python/31e_80A_PanoramixsPrediction.py
def isprime(a):
    for i in range(2,(a // 2) + 1):
        if a % i == 0:
            return False
    return True

def main():
    n, m = input().split()
    n, m = int(n), int (m)
    if not isprime(m):
        print("NO")
    else:
        flag = 0
        for i in range(n+1, m):
            if isprime(i):
                flag = 1
                break
        if flag == 0:
            print("YES")
        else:
            print("NO")

if __name__ == "__main__":
    main()